### ISSUE TITLE
Migrate Modules to Github Actions

### ISSUE THREAD

**Issue by jameesjohn** [oppia/oppiabot#277]
The Oppiabot project is currently in a hybrid state comprising of both Github Actions and Probot.

This issue aims to migrate the features to GitHub actions and track the progress of doing that.
To be able to contribute to this, you need to read the migration guide which can be found [here](https://github.com/oppia/oppiabot/wiki/Migrating-to-Github-Actions).

Sample PRs on migrating to Github Actions can be found [here](https://github.com/oppia/oppiabot/pull/276).

Available Modules to Migrate:
- [ ] apiForSheets.js
- [ ] checkBranchPush.js
- [ ] checkCriticalPr.js
- [ ] checkForNewCodeowners.js
- [ ] checkIssueAssignee.js
- [ ] checkMergeConflicts.js
- [ ] checkNewCron.js
- [ ] checkPullRequestBranch.js
- [ ] checkPullRequestJob.js
- [ ] checkPullRequestLabels.js
- [ ] checkPullRequestReview.js
- [ ] checkPullRequestTemplate.js
- [ ] ciChecks.js
- [ ] periodicChecks.js
- [ ] staleBuildChecks.js

In Progress:
- [ ] Wip/Branch Check @jameesjohn 

Done:

**Comment by nlok5923** [comment]
@jameesjohn  i would like to work on `checkIssueAssignee.js`

**Comment by jameesjohn** [comment]
Done @nlok5923. You can start working on it.

**Comment by khanjasir90** [comment]
@jameesjohn can i work on this checkPullRequestLabels.js

**Comment by vojtechjelinek** [comment]
@khanjasir90 Would you still be interested in this issue?

**Comment by khanjasir90** [comment]
Yes

**Comment by vojtechjelinek** [comment]
@khanjasir90 Done.

**Comment by SaurabhJamadagni** [comment]
Hi @vojtechjelinek @jameesjohn! Sorry for this very trivial question, but what are the technical requirements to be able to solve a task in this issue? This being my second issue ever I am a bit confused. I would really like to acquaint myself with the skills needed and contribute to this issue.

**Comment by vojtechjelinek** [comment]
You should know JavaScript and the GitHub Actions API. I also recommend reading through the migration guide https://github.com/oppia/oppiabot/wiki/Migrating-to-Github-Actions and the sample PR https://github.com/oppia/oppiabot/pull/276

**Comment by SaurabhJamadagni** [comment]
I see. I will read up on it. I would like to give it a try. @vojtechjelinek can you please assign me `staleBuildChecks.js`?

**Comment by vojtechjelinek** [comment]
@SaurabhJamadagni Done.

**Comment by SaurabhJamadagni** [comment]
Hi @vojtechjelinek, could you please assign me `checkPullRequestBranch.js` instead of `staleBuildChecks.js`? I am sorry for the inconvenience caused.

**Comment by vojtechjelinek** [comment]
@SaurabhJamadagni Sure

**Comment by SaurabhJamadagni** [comment]
Hi @vojtechjelinek! I am going through the migration guide and I have completed the 'Migrating A Feature' steps mentioned in the documentation. I am having trouble understanding a few steps in the 'Migrating Tests' section of the guide. Could you please help me understand the step '[Mocking](https://github.com/oppia/oppiabot/wiki/Migrating-to-Github-Actions#2-mocking)' mentioned? Do I have to make changes to the spec file? Thanks for sparing your time!

**Comment by vojtechjelinek** [comment]
@SaurabhJamadagni Yep, you need to modify the specs, because now the tests will probably fail. Let me know if you need more info.

**Comment by SaurabhJamadagni** [comment]
> @SaurabhJamadagni Yep, you need to modify the specs, because now the tests will probably fail. Let me know if you need more info.

@vojtechjelinek by looking at the reference PR, I understand that I will have to remove all instances of 'checkBranch' from all the spec files. Is there anything else I have to do next or is that the final step in the migration for the respective file? I am extremely sorry if these are trivial questions.

**Comment by vojtechjelinek** [comment]
> > @SaurabhJamadagni Yep, you need to modify the specs, because now the tests will probably fail. Let me know if you need more info.
> 
> @vojtechjelinek by looking at the reference PR, I understand that I will have to remove all instances of 'checkBranch' from all the spec files. Is there anything else I have to do next or is that the final step in the migration for the respective file? I am extremely sorry if these are trivial questions.

Yeah, then you will need to check if all the tests are still passing and if not you will need to modify the so that they pass.

**Comment by viveky45** [comment]
@jameesjohn i would like to work on apiForSheets.js

**Comment by viveky45** [comment]
@vojtechjelinek i would like to work on apiForSheets.js

**Comment by vojtechjelinek** [comment]
@viveky45 Done.

**Comment by viveky45** [comment]
@vojtechjelinek i was going through the migration guide but i have a liitle problem in the first two steps of migrating a module....i mean which event we have to change in first step and what entry we have to make in second step??

**Comment by coder1706** [comment]
Hi @vojtechjelinek, I would like to work on checkPullRequestReview.js. Could you assign it to me. Thanks!

**Comment by vojtechjelinek** [comment]
@coder1706 Done.

**Comment by coder1706** [comment]
Hi @vojtechjelinek, I have done up to step 5 mentioned in [Migrating a feature](https://github.com/oppia/oppiabot/wiki/Migrating-to-Github-Actions#migrating-a-feature), but could you help me understand the changes done in [actions_build/index.js](https://github.com/oppia/oppiabot/pull/276/files#diff-77ee5cd9161e9ef2b4100a824fa66aa7b59aff6ea8219bf6e26bbdb60a5b47af). Do I have to do these changes too?

**Comment by Ketan1502** [comment]
hi @vojtechjelinek, can u please assign me the ciChecks.js module to me?

**Comment by Ketan1502** [comment]
@vojtechjelinek can u please review the PR #300

**Comment by himanshusharma27** [comment]
I want to work on this issue . Please assign me this issue .
